backend/server.ts

```diff
@@ -19,53 +21,45 @@ app.use(cors());
 // Serve static files from the "Frontend" directory
 app.use(express.static('Frontend'));
 
-// Set up AWS S3
-const s3 = new AWS.S3({
-    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
-    region: 'us-east-2',
-});
-
 const storage = multer.memoryStorage(); // Store the file in memory
 const upload = multer({ storage: storage });
 
-app.get('/upload-page', (req, res) => {
+app.get('/upload-page', (req, res) => { // This is just for testing purposes
     res.sendFile(path.join(__dirname, '../Frontend/testwebsite.html'));
 });
 
-app.post('/upload', upload.single('packageContent'), async (req, res, next) => {
+app.post('/package', upload.single('packageContent'), async (req, res) => {
     try {
         if (!req.file) {
-            console.warn('No file provided in the upload.');
+            logger.info("No file provided in the upload.");
             return res.status(400).send('No file uploaded');
         }
 
-        const bucketName = process.env.AWS_S3_BUCKET_NAME;
+        const metadata = await apiPackage.extractMetadataFromZip(req.file.buffer);
+        const url = await apiPackage.getGithubUrlFromZip(req.file.buffer);
+        const githubInfo = apiPackage.parseGitHubUrl(url); 
 
-        if (!bucketName) {
-            console.error('Error: S3 bucket name not configured.');
-            return res.status(500).send('S3 bucket name not configured.');
+        if (!githubInfo) {
+            logger.info("Invalid GitHub repository URL.");
+            return res.status(400).send('Invalid GitHub repository URL.');
```

Comment:
you'd have to check this but it could also be a npm url. So you'd have to handle that case as well.